### PR TITLE
Add support for linting less files with lessc

### PR DESCRIPTION
**Pull request by zanona**

### CODE REVIEW

File: ale_linters/less/lessc.vim [hunk 1]
```diff
@@ -0,0 +1,36 @@
+" Author: zanona <https://github.com/zanona>, w0rp <devw0rp@gmail.com>
+" Description: This file adds support for checking Less code with lessc.
+
+call ale#Set('less_lessc_options', '')
+
+function! ale_linters#less#lessc#GetCommand(buffer) abort
+    return 'lessc'
+          \ . ' --no-color --lint '
+          \ . ale#Var(a:buffer, 'less_lessc_options')
+          \ . ' %t'
```
Comment (w0rp) [inline ale_linters/less/lessc.vim]: Align the line line continuations a little differently. Look through the rest of the codebase for examples.

File: ale_linters/less/lessc.vim [hunk 2]
```diff
@@ -0,0 +1,36 @@
+" Author: zanona <https://github.com/zanona>, w0rp <devw0rp@gmail.com>
```
Comment (w0rp) [inline ale_linters/less/lessc.vim]: You can just credit yourself for this.

File: ale_linters/less/lessc.vim [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+" Description: This file adds support for checking Less code with lessc.
+
+call ale#Set('less_lessc_options', '')
+
+function! ale_linters#less#lessc#GetCommand(buffer) abort
+    return 'lessc'
+          \ . ' --no-color --lint '
+          \ . ale#Var(a:buffer, 'less_lessc_options')
```
Comment (w0rp) [inline ale_linters/less/lessc.vim]: You might want to only add an extra space on the left if the options aren't empty.

**Comment by zanona** [comment]
Thanks, @w0rp, I have sorted out those issues. Hopefully, those will be ok now.
I have also added a patch https://github.com/w0rp/ale/pull/1103/commits/4bc31fcd18bda03f0de516be0380d6885cdd3d9b to allow properly importing files on less resolving to the current file path.

I wasn't sure if Ale provided any convenience method to retrieve the current file directory so I used `expand('%:p:h')` but please let me know if there's something else we could use.

Thanks for your help.

File: ale_linters/less/lessc.vim [hunk 4]
```diff
@@ -0,0 +1,36 @@
+" Author: zanona <https://github.com/zanona>
+" Description: This file adds support for checking Less code with lessc.
+
+call ale#Set('less_lessc_options', '')
+
+function! ale_linters#less#lessc#GetCommand(buffer) abort
+    return 'cat %t | lessc'
```
Comment (w0rp) [inline ale_linters/less/lessc.vim]: Remove `'cat %t |'` here. That won't work on Windows. If you don't put `%t` in the command, ALE will automatically put `< temporary_file` on the end for you.

**Comment by zanona** [comment]
Nice! thanks for your help.

**Comment by zanona** [comment]
Cool. Thanks, now I need to learn about Vader tests which are new to me :)
No problem tho, that's a good experience.
Would you have any recommendations on what to test? I tried to follow the `tests` directory but could find any linter specific test to use as a guide? Thanks again @w0rp.

**Comment by w0rp** [comment]
Have a look in the tests in `test/command_callback` and `test/handler`. There are a few for various linters.

**Comment by w0rp** [comment]
I made the following changes and merged this.

1. I added support for running the version of `lessc` from `node_modules,` like other Node linters.
2. I escaped some of the options better.
3. I made the handler add the `'filename'` key to the items for problems in other files, so you can jump to other files from the loclist or quickfix list.
4. I added tests for everything.

Cheers! :beers:

**Comment by zanona** [comment]
Wow! Amazing! Thanks a lot.

**Pull request merged**